openmdao/core/system.py

```diff
@@ -2565,3 +2608,23 @@ def _list_states(self):
             states.extend(subsys._list_states())
 
         return states
+
+    def add_recorder(self, recorder):
+        """
+        Add a recorder to the driver.
+
+        Parameters
+        ----------
+        recorder : <BaseRecorder>
+           A recorder instance.
+        """
+        self._rec_mgr.append(recorder)
+
+    def record_iteration(self):
+        """
+        Record an iteration of the current System.
+        """
+        metadata = create_local_meta(self.pathname)
+        # Send the calling method name into record_iteration, e.g. 'solve_nonlinear'.
+        self._rec_mgr.record_iteration(self, metadata, method=inspect.stack()[2][3])
```

Comment:
We should be able to keep track of the calling method using our own stack.  I suppose we can fix this later when we start concentrating on performance optimization.  There are currently a couple of other places (add_input/add_output) where we're doing it that will also have to be fixed.  Just to give you an idea of how slow this is, I profiled propulsor.py, and inspect.stack(), which is called once for each add_input and add_output, is taking up nearly 30% of the total execution time.